### PR TITLE
add lib_num_to_formated_str

### PR DESCRIPTION
**Pull request by nuggreat**
a library for changing numbers into formatted strings

### CODE REVIEW

File: doc/lib_num_to_formated_str.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
+// This file is distributed under the terms of the MIT license, (c) the KSLib team
+
+## lib_number_to_formated_string.ks
```
Comment (scimas) [inline doc/lib_num_to_formated_str.md]: Should be
```suggestion
## lib_num_to_formated_string.ks
```

File: doc/lib_num_to_formated_str.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+``lib_number_to_formated_string.ks`` provides several functions for changing numbers (scalers) into strings with specified formats
```
Comment (scimas) [inline doc/lib_num_to_formated_str.md]: Same
```suggestion
``lib_num_to_formated_string.ks`` provides several functions for changing numbers (scalers) into strings with specified formats
```

File: examples/example_lib_num_to_formated_str.ks [hunk 3]
```diff
@@ -0,0 +1,31 @@
+// This file is distributed under the terms of the MIT license, (c) the KSLib team
+
+RUN lib_number_to_formated_string.
```
Comment (scimas) [inline examples/example_lib_num_to_formated_str.ks]:
```suggestion
RUN lib_num_to_formated_str.
```

File: library/lib_num_to_formated_str.ks [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+  PARAMETER num,//number to format,
+  unit IS "".//unit of number
+  IF num = 0 {
+    RETURN padding(num,1,3) + "  " + unit.
+  } ELSE {
+    LOCAL powerOfTen IS MAX(MIN(FLOOR(LOG10(ABS(num))),26),-24).
+    LOCAL SIfactor IS FLOOR(powerOfTen / 3).
+    LOCAL trailingLength IS 3 - (powerOfTen - SIfactor * 3).
+    LOCAL prefix IS lib_formating_lex["siPrefixList"][SIfactor + 8].
+    RETURN padding(num/1000^SIfactor,1,trailingLength,TRUE,1) + prefix + unit.
```
Comment (scimas) [inline library/lib_num_to_formated_str.ks]:
```suggestion
    RETURN padding(num/1000^SIfactor,1,trailingLength,TRUE,0) + prefix + unit.
```
Should perform normal rounding per significant figures rules.

**Comment by nuggreat** [comment]
that should fix the issues you raised thanks for looking it over

File: examples/example_lib_num_to_formated_str.ks [hunk 5]
```diff
@@ -0,0 +1,31 @@
+// This file is distributed under the terms of the MIT license, (c) the KSLib team
+
+RUN lib_number_to_formated_str.
```
Comment (scimas) [inline examples/example_lib_num_to_formated_str.ks]:
```suggestion
RUN lib_num_to_formated_str.
```
Still needs changes, even in docs.

**Comment by nuggreat** [comment]
That should fix it for real now sorry about that

File: library/lib_num_to_formated_str.ks [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+
+  LOCAL place IS 1.
+  FOR modValue IN lib_formating_lex["timeModList"] {
+    LOCAL returnValue IS MOD(localTime,modValue).
+    returnList:ADD(returnValue).
+
+    SET localTime TO FLOOR(localTime / modValue).
+    IF localTime = 0 { BREAK. }
+    SET place TO place + 1.
+    IF place = places { BREAK. }
```
Comment (scimas) [inline library/lib_num_to_formated_str.ks]: `places` is undefined.

**Comment by nuggreat** [comment]
sorry about all the work i am generating for you i through i had this working this should be the last one

**Pull request merged**